src/include/processor/operator/persistent/index_builder.h

```diff
@@ -25,13 +25,28 @@ namespace processor {
 
 const size_t SHOULD_FLUSH_QUEUE_SIZE = 32;
 
+constexpr size_t BUFFER_SIZE = 64;
```

Comment:
I feel a little hypocritical saying this given that `in_mem_hash_index.h` has a very similar variable, but particularly since there is another variable called `BUFFER_SIZE` and the namespace alone isn't really enough to explain what they're for, maybe they should both be named, e.g. `INDEX_BUFFER_SIZE` and `WARNING_DATA_BUFFER_SIZE`. That or we should add a static constexpr `capacity` function to StaticVector so that the buffer size can be inlined and you can just call `OptionalWarningDataBuffer::capacity()` if you need to reference it.
That said, I don't actually see anywhere it's being referenced. The IndexBuffer was using a `BUFFER_SIZE` constant because there's a spot where a temporary array of the same length was being initialized, and even there it's something which should be removed since the `hashes` array in `InMemHashIndex::append` is no longer necessary.

Having two almost identically named values of the same type that are used in the same part of the code but which have very different values seems like it will be too easy for someone to accidentally use the wrong one.